Automattic/jetpack
# Search: add last indexed date for record meter

This very small PR adds the `lastIndexedDate` selector & props to record meter. 

#### Changes proposed in this Pull Request:

* adds a new selector in store/selectors/site-stats.js
* uses the selector to retrieve the value in dashboard/index.jsx & passes the value as a new prop to the RecordMeter component

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* On a site with an active Jetpack Search subscription visit `/wp-admin/admin.php?page=jetpack-search&features=record-meter` and check the last indexed date is displayed below the record meter. 

![image](https://user-images.githubusercontent.com/30754158/160010870-c63f6e8a-1cb2-45ef-abf9-ffe74dcfb833.png)
